### DISCUSSION TITLE
Can netbootxyz use ISOs hosted on LAN?

### DISCUSSION THREAD

**Discussion by oxwivi** [antonym/netboot.xyz#1382] [Q&A]
Instead of downloading over the Internet every time, is there a way I can point netbootxyz to my cache on LAN (whether it's TFTP, SMB, NFS or something else) and use that instead? Or does that necessarily entail self-hosting the complete netbootxyz server?

I was hoping for using the generic PXE boot file and point it to the ISOs on file server while booting into it. I didn't find anything similar in the docs, so if such functionality is not present, I hope this can be taken as a suggestion/feature request.

**Comment by erdnuesse** [comment]
Yes. If you use the docker container, local location is under /assets. 
However, you seem to be able to set them in a web interface. 
More info [here](https://a-t-engineering.com/en/pxe-network-boot-using-netboot-xyz-under-docker/).
source: currently researching, not implemented myself yet.

**Comment by oxwivi** [comment]
Ah, Docker? I was hoping to have netbootxyz on my OpenWrt router, and at the same time point to TFTP/NFS cache on it for the ISOs...

**Comment by erdnuesse** [comment]
in that case I would try mounting your local share to /assets, then use the built-in download to local option and changing the configs mentioned in the link.

However, think of the security implications, running a service on an inet facing device that has local storage access.
At the very least reduce the access to that share under the least privileged access principles, RBAC and port filtering. That's how you could do it. 

However and unsolicited advisory: IMO,  better to run it internal only and point to the tftp via DHCP option on your router. that's how I did it,  keeping the services on the internet facing device as few as possible, and just as much as necessary. 
You should be able to find the necessary options in the docs. 

instead of running on the router, you'd need a separate device running docker or kvm. Any proxmox host, raspberry pi or microserver could do. if you don't have that luxury, maybe running a Netboot isn't the highest priority, but your priorities are your discretion ofc.

**Reply by oxwivi** [reply]
That's great and all, but I suppose there's no way around using Docker for doing the `/assets` thing? Since I want to use an OpenWrt box itself as TFTP server, Docker seems to be a bit much. But I suppose if I'm doing it on AP, it won't be too bad.
